File: Client_Root/Client/Assets/Scripts/Network/Messages/ToLobby/JoinLobbyToL.cs
using FlatBuffers;

public class JoinLobbyToL : IMessage
{
    public const ushort MESSAGE_ID = MessageID.JoinLobbyToL_ID;

    public string m_strPlayerKey;
    public int m_nAuthKey;

    public override ushort GetID()
    {
        return MESSAGE_ID;
    }

    public override IMessage Clone()
    {
        return null; 
    }

    public override byte[] Serialize()
    {
        var playerKey = m_Builder.CreateString(m_strPlayerKey);

        FBS.JoinLobbyToL.StartJoinLobbyToL(m_Builder);
        FBS.JoinLobbyToL.AddPlayerKey(m_Builder, playerKey);
        FBS.JoinLobbyToL.AddAuthKey(m_Builder, m_nAuthKey);
        var data = FBS.JoinLobbyToL.EndJoinLobbyToL(m_Builder);

        m_Builder.Finish(data.Value);

        return m_Builder.SizedByteArray();
    }

    public override bool Deserialize(byte[] bytes)
    {
        var buf = new ByteBuffer(bytes);

        var data = FBS.JoinLobbyToL.GetRootAsJoinLobbyToL(buf);

        m_strPlayerKey = data.PlayerKey;
        m_nAuthKey = data.AuthKey;

        return true;
    }
}

File: Server_Root/Server/Server/Messages/ToLobby/JoinLobbyToL.cpp
#include "stdafx.h"
#include "JoinLobbyToL.h"
#include "../../../FBSFiles/JoinLobbyToL_generated.h"

JoinLobbyToL::JoinLobbyToL()
{
}

JoinLobbyToL::~JoinLobbyToL()
{
}

unsigned short JoinLobbyToL::GetID()
{
	return MESSAGE_ID;
}

IMessage* JoinLobbyToL::Clone()
{
	return NULL;
}

const char* JoinLobbyToL::Serialize(int* pLength)
{
	auto playerKey = m_Builder.CreateString(m_strPlayerKey);

	FBS::JoinLobbyToLBuilder data_builder(m_Builder);
	data_builder.add_PlayerKey(playerKey);
	data_builder.add_AuthKey(m_nAuthKey);
	auto data = data_builder.Finish();

	m_Builder.Finish(data);

	*pLength = m_Builder.GetSize();

	return (char*)m_Builder.GetBufferPointer();
}

bool JoinLobbyToL::Deserialize(const char* pChar)
{
	auto data = flatbuffers::GetRoot<FBS::JoinLobbyToL>((const void*)pChar);

	m_strPlayerKey = data->PlayerKey()->str();
	m_nAuthKey = data->AuthKey();

	return true;
}